### ISSUE TITLE
TODO item fails to appear in TODO list

### ISSUE THREAD

**Issue by daFreeMan** [rubberduck-vba/Rubberduck#563]
I added a TODO item to this snip of code:

```
If SourceField = "Q11" Then                 'set the column numbers for formatting later
  Q11Col = I
ElseIf SourceField = "Q16" Then
  Q16Col = I
End If
 
'TODO: I know what set of columns we need to keep, just need to read that set from the DB & see if this column is in it.
DestCol = GetGenericConfigInfo(OutData, "ColumnXlate", SourceField, 1, XlateCol)
If Len(DestCol) > 0 Then
  SQLString = SQLString & SourceField & ", "
      'ValueString will get an invalid column "13-20" this will be translated into the appropriate column in TransferData()
  ValuesString = ValuesString & GetGenericConfigInfo(OutData, "DestColumnNames", DestCol, 1, 2) & ", "
  DestColumns = DestColumns & DestCol & ","
End If
I = I + 1
```
then refreshed the TODO list, and it did not appear in the list. As shown below, other items have been picked up as both TODO and NOTES.

This fails even after saving the Access DB, closing Access & reloading.

![rd](https://cloud.githubusercontent.com/assets/11889733/7846837/f7935e16-048d-11e5-9569-bafbd5b83590.PNG)

**Comment by ckuhn203** [comment]
@daFreeMan Any chance you can include your `rubberduck.config` file? It's under \AppData\Roaming\

**Comment by daFreeMan** [comment]
One `Rubberduck.config` as requested:

```<?xml version="1.0" encoding="utf-8"?>
<Configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <UserSettings>
    <ToDoListSettings>
      <ToDoMarkers>
        <ToDoMarker Text="NOTE:" Priority="Low" />
        <ToDoMarker Text="TODO:" Priority="Normal" />
        <ToDoMarker Text="BUG:" Priority="High" />
      </ToDoMarkers>
    </ToDoListSettings>
    <CodeInspectionSettings>
      <CodeInspections>
        <CodeInspection Name="Member '{0}' is implicitly Public" Severity="Suggestion" InspectionType="MaintainabilityAndReadabilityIssues" />
        <CodeInspection Name="Function '{0}' returns no value" Severity="Warning" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Use of obsolete Call statement" Severity="Warning" InspectionType="LanguageOpportunities" />
        <CodeInspection Name="Use of obsolete Global access modifier" Severity="Suggestion" InspectionType="LanguageOpportunities" />
        <CodeInspection Name="Use of obsolete Let statement" Severity="Warning" InspectionType="LanguageOpportunities" />
        <CodeInspection Name="Use of obsolete type hints in identifier '{0}'" Severity="Hint" InspectionType="LanguageOpportunities" />
        <CodeInspection Name="Potentially confusing implicit array lower bound" Severity="Hint" InspectionType="MaintainabilityAndReadabilityIssues" />
        <CodeInspection Name="Parameter '{0}' can be passed by value" Severity="DoNotShow" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Parameter '{0}' is not used" Severity="Hint" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Variable '{0}' is never used" Severity="Hint" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Variable '{0}' is never assigned" Severity="Error" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Parameter '{0}' is passed ByRef implicitly" Severity="Suggestion" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Function '{0}' returns an implicit Variant" Severity="Suggestion" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Instruction contains multiple declarations" Severity="Suggestion" InspectionType="MaintainabilityAndReadabilityIssues" />
        <CodeInspection Name="Use of obsolete Rem comment syntax" Severity="Suggestion" InspectionType="LanguageOpportunities" />
        <CodeInspection Name="Option Explicit is not specified" Severity="Warning" InspectionType="CodeQualityIssues" />
        <CodeInspection Name="Variable '{0}' is implicitly Variant" Severity="Suggestion" InspectionType="LanguageOpportunities" />
      </CodeInspections>
    </CodeInspectionSettings>
  </UserSettings>
</Configuration>
```

Also, I tried today to change `TODO` to `BUG` and got the same result after a refresh - nothing.

**Comment by Hosch250** [comment]
I tested this issue with the following data:

    ' NOTE:
    ' BUG:
    ' TODO:
    'BUG
    '  TODO

Only items with a single space after the comment character work.

**Comment by daFreeMan** [comment]
Sadly, a space after the comment character doesn't help, either @Hosch250:

![rd](https://cloud.githubusercontent.com/assets/11889733/7866651/e89aa732-053e-11e5-831c-8afcc086d0c6.PNG)

**Comment by ckuhn203** [comment]
I suspect a parsing error, and given that this doesn't *have* to rely on the parser, I'm considering breaking that dependencies. Parsing these out via the VBIDE library isn't a difficult thing to do.

**Comment by retailcoder** [comment]
Comments *are* parsed out of the VBIDE API... they are parsed in `RubberduckParser` so that code that needs them can get them from the parse results.

**Comment by ckuhn203** [comment]
I didn't realize that. I thought you had changed it so that they were actually coming out of the parser proper.

**Comment by retailcoder** [comment]
Nope. Grammar ignores them altogether :wink:

**Comment by ckuhn203** [comment]
Okay, well... it's just a *different kind* of parsing error then. lol. Will look into it as soon as I have a moment @daFreeMan.

**Comment by daFreeMan** [comment]
It appears this is fixed as of v1.4.3.

Tested it with:
- 'Bug:
- 'BUG:
- 'Note:
- 'NOTE:
- 'Todo:
- 'TODO:
